core/mo-services-impl/nmf-platform-generic-impl/src/main/java/esa/mo/platform/impl/provider/gen/GPSProviderServiceImpl.java

```diff
@@ -770,19 +860,76 @@ public void getBestXYZSentence(GetBestXYZSentenceInteraction interaction)
   public void getTIMEASentence(GetTIMEASentenceInteraction interaction)
       throws MALInteractionException, MALException
   {
+    String resp = null;
     if (!adapter.isUnitAvailable()) { // Is the unit available?
-      throw new MALInteractionException(
-          new MALStandardError(PlatformHelper.DEVICE_NOT_AVAILABLE_ERROR_NUMBER, null));
+      if(isTLEFallbackEnabled){
+        resp = HelperTLE.generateTIMEALog(utcOffset);
+      }
+      else {
+        throw new MALInteractionException(
+                new MALStandardError(PlatformHelper.DEVICE_NOT_AVAILABLE_ERROR_NUMBER, null));
+      }
     }
     interaction.sendAcknowledgement();
     try {
-      String resp = adapter.getTIMEASentence();
-      interaction.sendResponse(resp);
+      interaction.sendResponse(resp == null ? adapter.getTIMEASentence() : resp);
     } catch (IOException e) {
       interaction
           .sendError(new MALStandardError(PlatformHelper.DEVICE_NOT_AVAILABLE_ERROR_NUMBER, null));
       e.printStackTrace();
     }
   }
 
+  /**
+   * Checks the fixQquality on the GNSS position.
+   * @return true if a fixed position has been established, false otherwise
+   */
+  private boolean isPositionFixed() {
+    boolean isFixed = false;
+    try {
+      if(adapter.getCurrentPosition().getExtraDetails().getFixQuality() > 0){
+        isFixed = true;
+      }
+    } catch (NullPointerException e)
+    {
+      LOGGER.warning("Could not receive a fixed position: " + e.getMessage());
+    }
+    return isFixed;
+  }
+
+  private Position getTLEPropagatedPosition()
+  {
+    Calendar targetDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    SpacecraftState state = getSpacecraftState(targetDate);
+
+    // Converting to geodetic lat/lon/alt
+    Frame ecf = FramesFactory.getITRF(IERSConventions.IERS_2010,true);
+    OneAxisEllipsoid earth = new OneAxisEllipsoid(Constants.WGS84_EARTH_EQUATORIAL_RADIUS, Constants.WGS84_EARTH_FLATTENING, ecf);
+    GeodeticPoint satLatLonAlt = earth.transform(state.getPVCoordinates().getPosition(), FramesFactory.getEME2000(),
+            state.getDate());
+
+    PositionExtraDetails extraDetails = new PositionExtraDetails();
+    extraDetails.setPositionSource(PositionSourceType.TLE);
+    extraDetails.setUtc(new Time(targetDate.getTimeInMillis()));
```

Comment:
Should we also initialize all other fields with zeros (eg hdop, undulation, etc)?